app/src/main/java/com/example/tasky/common/domain/util/StringToDateExtension.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.tasky.common.domain.util
+
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Date
+import java.util.Locale
+
+fun Long.convertMillisToMonth(): String {
+    val dateFormat = SimpleDateFormat("MMMM", Locale.getDefault())
+    return dateFormat.format(Date(this))
+}
+
+fun Long.convertMillisToDate(): String {
+    // Create a calendar instance in the default time zone
+    val calendar = Calendar.getInstance().apply {
+        timeInMillis = this@convertMillisToDate
+        // Adjust for the time zone offset to get the correct local date
+        val zoneOffset = get(Calendar.ZONE_OFFSET)
+        val dstOffset = get(Calendar.DST_OFFSET)
+        add(Calendar.MILLISECOND, -(zoneOffset + dstOffset))
+    }
+    // Format the calendar time in the specified format
+    val sdf = SimpleDateFormat("MMM dd, yyyy", Locale.US)
```

Comment:
The more modern way to format dates is by using the Java LocalDateTime API, since `SimpleDateFormat` is not thread-safe (not an issue here, but as soon as you'd share this `sdf` instance across threads you can run into issues). This API is also quite simpler than what you have here to get this working with time zones and all:
```
val dateTime = ZonedDateTime.ofInstant(
    Instant.ofEpochMilli(millis),
    ZoneId.of("UTC")
)
val formattedDateTime = DateTimeFormatter
    .ofPattern("dd MMM yyyy")
    .format(dateTime)
```